File: app/src/main/java/habibkazemi/ir/lingualyrics_android/vo/Lyric.java
package habibkazemi.ir.lingualyrics_android.vo;

import android.arch.persistence.room.Entity;


import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

@Entity(primaryKeys = {"id"})
public class Lyric {

	private Integer id;

	@SerializedName("lyric_text")
	@Expose
	private String lyricText;

	@SerializedName("artist")
	@Expose
	private String artist;

	@SerializedName("success")
	@Expose
	private boolean success;

	@SerializedName("album")
	@Expose
	private String album;

	@SerializedName("cover_art_image_url")
	@Expose
	private String coverArtImageUrl;

	@SerializedName("title")
	@Expose
	private String title;

	@SerializedName("url")
	@Expose
	private String url;

	public void setLyricText(String lyricText){
		this.lyricText = lyricText;
	}

	public String getLyricText(){
		return lyricText;
	}

	public void setArtist(String artist){
		this.artist = artist;
	}

	public String getArtist(){
		return artist;
	}

	public void setSuccess(boolean success){
		this.success = success;
	}

	public boolean isSuccess(){
		return success;
	}

	public void setAlbum(String album){
		this.album = album;
	}

	public String getAlbum(){
		return album;
	}

	public void setCoverArtImageUrl(String coverArtImageUrl){
		this.coverArtImageUrl = coverArtImageUrl;
	}

	public String getCoverArtImageUrl(){
		return coverArtImageUrl;
	}

	public void setTitle(String title){
		this.title = title;
	}

	public String getTitle(){
		return title;
	}

	public void setUrl(String url){
		this.url = url;
	}

	public String getUrl(){
		return url;
	}

	public Integer getId() {return this.id;}

	public void setId(int id) {this.id = id;}

	@Override
 	public String toString(){
		return 
			"Lyric{" +
			"lyric_text = '" + lyricText + '\'' + 
			",artist = '" + artist + '\'' + 
			",success = '" + success + '\'' + 
			",album = '" + album + '\'' + 
			",cover_art_image_url = '" + coverArtImageUrl + '\'' + 
			",title = '" + title + '\'' + 
			",url = '" + url + '\'' + 
			"}";
		}
}

File: app/src/main/java/ir/habibkazemi/lingualyrics/vo/Lyric.kt
package ir.habibkazemi.lingualyrics.vo

import android.arch.persistence.room.Entity


import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

@Entity(primaryKeys = ["id"])
class Lyric {

    var id: Int? = null
        private set

    @SerializedName("lyric_text")
    @Expose
    var lyricText: String? = null

    @SerializedName("artist")
    @Expose
    var artist: String? = null

    @SerializedName("source")
    @Expose
    var source: String? = null

    @SerializedName("success")
    @Expose
    var success: Boolean = false

    @SerializedName("album")
    @Expose
    var album: String? = null

    @SerializedName("cover_art_image_url")
    @Expose
    var coverArtImageUrl: String? = null

    @SerializedName("title")
    @Expose
    var title: String? = null

    @SerializedName("url")
    @Expose
    var url: String? = null

    fun setId(id: Int) {
        this.id = id
    }

    override fun toString(): String {
        return """Lyric{lyric_text = '$lyricText',
            |artist = '$artist',
            |success = '$success',
            |source = '$source',
            |album = '$album',
            |cover_art_image_url = '$coverArtImageUrl',
            |title = '$title',
            |url = '$url'}"""
            .trimMargin()
    }
}